host/class/cdc/usb_host_cdc_acm/cdc_host_parsing.c

```diff
@@ -0,0 +1,214 @@
+/*
+ * SPDX-FileCopyrightText: 2024 Espressif Systems (Shanghai) CO LTD
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdbool.h>
+#include <string.h>
+#include "esp_check.h"
+#include "esp_log.h"
+#include "usb/usb_helpers.h"
+#include "usb/usb_types_cdc.h"
+#include "usb_cdc_acm_host_parsing.h"
+
+// CDC devices often implement Interface Association Descriptor (IAD). Parse IAD only when
+// bDeviceClass = 0xEF (Miscellaneous Device Class), bDeviceSubClass = 0x02 (Common Class), bDeviceProtocol = 0x01 (Interface Association Descriptor),
+// or when bDeviceClass, bDeviceSubClass, and bDeviceProtocol are 0x00 (Null class code triple), as per https://www.usb.org/defined-class-codes, "Base Class 00h (Device)" section
+// @see USB Interface Association Descriptor: Device Class Code and Use Model rev 1.0, Table 1-1
+#define USB_SUBCLASS_NULL          0x00
+#define USB_SUBCLASS_COMMON        0x02
+#define USB_PROTOCOL_NULL          0x00
+#define USB_DEVICE_PROTOCOL_IAD    0x01
+
+static const char *TAG = "cdc_acm_parsing";
+
+/**
+ * @brief Check if the required interface is CDC compliant
+ *
+ * @param[in] device_desc Pointer to Device descriptor
+ * @param[in] config_desc Pointer do Configuration descriptor
+ * @param[in] intf_idx    Index of the required interface
+ * @return true  The required interface is CDC compliant
+ * @return false The required interface is NOT CDC compliant
+ */
+static bool cdc_parse_is_cdc_compliant(const usb_device_desc_t *device_desc, const usb_config_desc_t *config_desc, uint8_t intf_idx)
```

Comment:
Here I deliberately omitted the 'ACM' part. Beginning of this year, we were designing support for CDC-ECM subclass and we decided that it will be implemented inside this driver. And the driver will be renamed (or aliased?) to `usb_host_cdc`.

So the new naming has `cdc_*` prefix for functions that should work for all subclasses and `cdc_acm_*` for functions that are specific to the ACM subclass.

The other naming comments make sense to me, will update